### PR TITLE
Test `clean` task in CI

### LINKED ISSUE #413: Improve cleaning the project
Build the project with `./gradlew assemble`.
Then, after running `./gradlew clean`, there are still many build artifacts lying around, e.g. nearly 2000 `.class` files and javadoc HTML files.
If there is a different task to run to clean all files, document it. (I tried `cleanTest` and `cleanBuild` and they don't clean these files.)
If no such task exists, improve the `clean` task.

Note that there are a few test tasks that use Makefiles. Improve those by including a `clean` target and make sure these are actually run.

### PR DESCRIPTION
**Pull request by Ao-senXiong**
Fixes #413

### CODE REVIEW

File: checker/bin-devel/test-cftests-junit.sh [hunk 1]
```diff
@@ -13,3 +13,6 @@ source "$SCRIPTDIR"/clone-related.sh
 # Adding --max-workers=1 to avoid random failures in Github Actions. An alternative solution is to use --no-build-cache.
 # https://github.com/eisop/checker-framework/issues/849
 ./gradlew test -x javadoc -x allJavadoc --console=plain --warning-mode=all --max-workers=1
+
+# Test clean task
+./gradlew clean
```
Comment (wmdietl) [inline checker/bin-devel/test-cftests-junit.sh:18]:
```suggestion
./gradlew clean
./gradlew clean
```

You need to run it twice, here and in the other tasks, to actually see what happens when you run `clean` on a `clean` directory. Otherwise, you again don't test what happens then.
The change just to `issue265/Makefile` is not enough.

This also doesn't really test that the directory is clean afterwards... but checking `git status` doesn't help, as most temporary files are being ignored.

Comment (wmdietl) [reply checker/bin-devel/test-cftests-junit.sh:18]: Ah, the change to `issue265/Makefile` is enough on CI. Let's run clean twice only here.

In `docs/examples/BazelExample/Makefile` can you change `bazelisk clean` to `-bazelisk clean` so that bazelisk isn't required for cleaning?

**Comment by wmdietl** [comment]
The problem on Windows is the `../../../gradlew clean` which appears in one of the Makefiles.

File: checker/bin-devel/test-cftests-junit.sh (hunk 1 above)
Comment (Ao-senXiong) [reply checker/bin-devel/test-cftests-junit.sh:18]: Added!

**Comment by Ao-senXiong** [comment]
> The problem on Windows is the `../../../gradlew clean` which appears in one of the Makefiles.

Thanks!

File: docs/examples/lombok/Makefile [hunk 2]
```diff
@@ -15,8 +15,7 @@ all:
 	  && grep -qF "Foo.java:12: error: [assignment.type.incompatible] incompatible types in assignment." Out.txt \
 	  && grep -qF "y = null; // error" Out.txt) \
 	 || (echo "===== start of Out.txt =====" && cat Out.txt && echo "===== end of Out.txt =====" && false)
-endif
-
 clean:
-	../../../gradlew clean
+	 ../../../gradlew clean
```
Comment (Ao-senXiong) [inline docs/examples/lombok/Makefile:19]:
```suggestion
	../../../gradlew clean
```

**Comment by Ao-senXiong** [comment]
> The problem on Windows is the `../../../gradlew clean` which appears in one of the Makefiles.

Hi @wmdietl, do you have any idea how to fix the problem? I did [0b1e15c](https://github.com/eisop/checker-framework/pull/896/commits/0b1e15c7d43af8bef0abf12f4c4ff944e3d1e901) this for windows but it looks like it is taking forever to run test for windows and fails eventually.

File: docs/examples/errorprone/Makefile [hunk 3]
```diff
@@ -7,6 +7,12 @@ all:
...
 
+ifeq ($(OS),Windows_NT)
+    GRADLEW_CMD := ..\..\..\gradlew.bat
+else
+    GRADLEW_CMD := ../../../gradlew
+endif
+
 clean:
-	../../../gradlew clean
+	$(GRADLEW_CMD) clean
```
Comment (wmdietl) [inline docs/examples/errorprone/Makefile:17]: How about undoing these changes and just using `- ../../...` linke in the `all` target above?
I assume that running all on Windows similarly fails at the moment.
For the `clean` tasks it is fine to silently not work, e.g. if a program isn't installed.

Comment (Ao-senXiong) [reply docs/examples/errorprone/Makefile:17]: I remembered I did it here [17b2897](https://github.com/eisop/checker-framework/pull/896/commits/17b2897b68fd2a2cd93c63d05356dce145465816) but it did not work. But let me try again.

File: checker/bin-devel/test-cftests-nonjunit.sh [hunk 4]
```diff
@@ -17,3 +17,6 @@ source "$SCRIPTDIR"/clone-related.sh
 # Moved example-tests out of all tests because it fails in
 # the release script because the newest maven artifacts are not published yet.
 ./gradlew :checker:exampleTests -x javadoc -x allJavadoc --console=plain --warning-mode=all
+
```
[the remark this answers is not in the document]
Comment (Ao-senXiong) [reply checker/bin-devel/test-cftests-nonjunit.sh:20]: I think :checker:examplesTests are tests run makefiles https://github.com/eisop/checker-framework/blob/304e999c7c3ca727f9e1eec7349cb42cf79fe862/checker/build.gradle#L305

[the remark this answers is not in the document]
Comment (wmdietl) [reply checker/bin-devel/test-cftests-nonjunit.sh:20]: Can you elaborate a bit what your point is?
We already test running `./gradlew clean` in `junit` above. Do you want to clean at the end of every test? Or why is `:checker:examplesTests` relevant for whether we call `clean` here?

[the remark this answers is not in the document]
Comment (Ao-senXiong) [reply checker/bin-devel/test-cftests-nonjunit.sh:20]: I think Junit test doesn't contain makefile tests such as `exampleTests`, is that correct? Because `exampleTests` are the tests for makefiles and `./gradlew clean` will also run make clean for makefile tests. I just want to make it works after we actual run makefile tests in CI.

[the remark this answers is not in the document]
Comment (wmdietl) [reply checker/bin-devel/test-cftests-nonjunit.sh:20]: There are no tests that actually ensure that `clean` cleaned up everything - so you won't know whether this call to `clean` actually removed all files that were generated from `exampleTests`.
So what do you learn from calling `clean` here?

[the remark this answers is not in the document]
Comment (Ao-senXiong) [reply checker/bin-devel/test-cftests-nonjunit.sh:20]: okay, I will clean up them. Thanks!

File: checker/bin-devel/test-cftests-nonjunit.sh [hunk 5]
```diff
@@ -17,3 +17,6 @@ source "$SCRIPTDIR"/clone-related.sh
...
+# Test clean task
+./gradlew clean
```
Comment (Ao-senXiong) [inline checker/bin-devel/test-cftests-nonjunit.sh:22]:
```suggestion

```

File: checker/bin-devel/test-cftests-nonjunit.sh [hunk 6]
```diff
@@ -17,3 +17,4 @@ source "$SCRIPTDIR"/clone-related.sh
 # Moved example-tests out of all tests because it fails in
 # the release script because the newest maven artifacts are not published yet.
 ./gradlew :checker:exampleTests -x javadoc -x allJavadoc --console=plain --warning-mode=all
+
```
Comment (Ao-senXiong) [inline checker/bin-devel/test-cftests-nonjunit.sh:20]:
```suggestion
```

**Review by wmdietl — APPROVED** [review]
Thanks!
